.github/actions/install/openssl/action.yml

```
@@ -0,0 +1,19 @@
+name: Install OpenSSL
+description: Install and setup OpenSSL for linking and building test application
+inputs:
+  version:
+    description: The desired OpenSSL version to install
+    required: false
+    default: "3.0.0-beta2"
+runs:
+  using: composite
+  steps:
+    - run: |
+        cd /tmp
+        wget http://artfiles.org/openssl.org/source/openssl-${{ inputs.version }}.tar.gz
```

I have no issue with using a mirror however perhaps using GitHub is better for actions? 

```suggestion
        wget https://github.com/openssl/openssl/archive/refs/tags/openssl-${{ inputs.version }}.tar.gz
```

---

Makes sense. I updated links, however,  the versions are now referenced as `openssl-3.0.0-beta2` and `OpenSSL_1_1_1k`. I don't know whether there is an easy support for something like maps in action files.